src/conv/heur/metadata.cpp

```diff
@@ -0,0 +1,397 @@
+        "batchsize",    "conv_stride_h", "conv_stride_w", "fil_h",
+        "fil_w",        "in_channels",   "in_h",          "in_w",
+        "out_channels", "pad_h",         "pad_w"};
+    return gfx908_feature_names;
+  } else if (arch == "gfx906")
+
+  {
+    static const std::vector<std::string> gfx906_feature_names = {
+        "batchsize",    "conv_stride_h", "conv_stride_w", "fil_h",
+        "fil_w",        "in_channels",   "in_h",          "in_w",
+        "out_channels", "pad_h",         "pad_w"};
+    return gfx906_feature_names;
+  }
+
+  else {
+    MIOPEN_THROW(miopenStatusNotImplemented);
+  }
+}
+
+const std::unordered_map<int, std::string> &GetSolverMap(const Handle &handle) {
+  const auto &arch = handle.GetDeviceName();
+
+  if (arch == "gfx908")
+
+  {
+    static const std::unordered_map<int, std::string> gfx908_solver_map = {
+        {0, "ConvAsmImplicitGemmGTCDynamicBwdXdlops"},
+        {1, "ConvAsmImplicitGemmGTCDynamicFwdXdlops"},
+        {2, "ConvAsmImplicitGemmGTCDynamicWrwXdlops"},
+        {3, "ConvBinWinograd3x3U"},
+        {4, "ConvBinWinogradRxS"},
+        {5, "ConvBinWinogradRxSf2x3"},
+        {6, "ConvBinWinogradRxSf3x2"},
+        {7, "ConvMPBidirectWinograd<3-3>"},
+        {8, "ConvMPBidirectWinograd<4-3>"},
+        {9, "ConvMPBidirectWinograd<5-3>"},
+        {10, "ConvMPBidirectWinograd<6-3>"},
+        {11, "ConvWinograd3x3MultipassWrW<3-4>"},
+        {12, "ConvWinograd3x3MultipassWrW<3-5>"},
+        {13, "ConvWinograd3x3MultipassWrW<3-6>"},
+        {14, "gemm"}};
+    return gfx908_solver_map;
+  } else if (arch == "gfx906")
+
+  {
+    static const std::unordered_map<int, std::string> gfx906_solver_map = {
+        {0, "ConvAsmImplicitGemmV4R1DynamicBwd"},
+        {1, "ConvAsmImplicitGemmV4R1DynamicFwd"},
+        {2, "ConvAsmImplicitGemmV4R1DynamicFwd_1x1"},
+        {3, "ConvAsmImplicitGemmV4R1DynamicWrw"},
+        {4, "ConvBinWinograd3x3U"},
+        {5, "ConvBinWinogradRxS"},
+        {6, "ConvBinWinogradRxSf2x3"},
+        {7, "ConvBinWinogradRxSf3x2"},
+        {8, "ConvMPBidirectWinograd<3-3>"},
+        {9, "ConvMPBidirectWinograd<4-3>"},
+        {10, "ConvMPBidirectWinograd<5-3>"},
+        {11, "ConvMPBidirectWinograd<6-3>"},
+        {12, "ConvWinograd3x3MultipassWrW<3-4>"},
+        {13, "ConvWinograd3x3MultipassWrW<3-5>"},
+        {14, "ConvWinograd3x3MultipassWrW<3-6>"},
+        {15, "gemm"}};
+    return gfx906_solver_map;
+  }
+
+  else {
+    MIOPEN_THROW(miopenStatusNotImplemented);
+  }
+}
+
+const std::vector<float> &GetMu(const Handle &handle,
+                                const ProblemDescription &problem) {
+  const auto &arch = handle.GetDeviceName();
+  auto fwd = problem.direction.IsForward();
+  auto bwd = problem.direction.IsBackwardData();
+  auto wrw = problem.direction.IsBackwardWrW();
+
+  if (arch == "gfx908" && fwd)
+
+  {
+    static const std::vector<float> gfx908_fwd_mu = {
+        3.5623828968151168, 1.0490354202886407, 1.0490354202886407,
```

Comment:
WHAT'S THIS

Reply:
These are the mean and standard deviation vectors for the features from the dataset. Which we need to normalize the features everytime we use the model to do inference.